### PR TITLE
Add validations to our swagger input field generators

### PR DESCRIPTION
**Pull request by macrael**
Only thing left for this to be ready is more testing coverage, but I would love eyes on this before the end of the day

### CODE REVIEW

File: src/shared/JsonSchemaForm/JsonSchemaField.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    if (i === 3 || i === 6) {
+      normalizedPhone += '-';
+    }
+    normalizedPhone += onlyNums[i];
+  }
+  return normalizedPhone;
+};
+
+const configureTelephoneField = (swaggerField, props) => {
+  props.normalize = normalizePhone;
```
Comment (tinyels) [inline src/shared/JsonSchemaForm/JsonSchemaField.js]: Right now it is possible to leave an incomplete phone number, e.g. '8-4444' and get no warning. This results in a server error. We need the validators to produce helpful error messages to the user.

True story: 8-4444 was my childhood phone number.

File: src/shared/JsonSchemaForm/JsonSchemaField.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  return props;
+};
+
+const configureTextField = (swaggerField, props) => {
+  if (swaggerField.maxLength) {
+    props.validate.push(validator.maxLength(swaggerField.maxLength));
+  }
+  if (swaggerField.minLength) {
+    props.validate.push(validator.minLength(swaggerField.minLength));
+  }
```
Comment (tinyels) [inline src/shared/JsonSchemaForm/JsonSchemaField.js]: These validators also need to produce helpful messages for the user.

Comment (macrael) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: Where are you seeing them not display anything? They should say "Must be at least X characters long."

File: src/shared/JsonSchemaForm/JsonSchemaField.js (hunk 1 above)
Comment (macrael) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: Fixed by adding format: telephone to the telephone numbers in the 1299 form.

File: src/shared/JsonSchemaForm/index.jsx [hunk 3]
```diff
@@ -1,15 +1,14 @@
-import React, { Fragment } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
+import SchemaField from './JsonSchemaField';
 
-import { Field, reduxForm } from 'redux-form';
+import { isEmpty } from 'lodash';
```
[the remark this answers is not in the document]
Comment (macrael) [reply src/shared/JsonSchemaForm/index.jsx]: ✅

**Comment by macrael** [comment]
Re: "We probably want a general validation for swagger pattern attributes"

I'm going to add a validator that is really meant to catch issues in dev. I don't think there is a reasonable way for us to convert a regex into an understandable error message. I'll just have it print the example if it's violated, but I think that we will end up creating custom validators for the different patterns anyway and use the format field to denote them.

File: src/shared/JsonSchemaForm/JsonSchemaField.js (hunk 2 above)
Comment (tinyels) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: You are right, I am seeing them now. However, we should implement the [uswds styling](https://designsystem.digital.gov/components/form-controls/)
```
  <div class="usa-input-error">
    <label class="usa-input-error-label" for="input-error">Text input error</label>
    <span class="usa-input-error-message" id="input-error-message" role="alert">Helpful error message</span>
    <input id="input-error" name="input-error" type="text" aria-describedby="input-error-message">
  </div>
```

**Pull request merged**